File: javascript/score-ad.mjs
#!/usr/bin/env node
/**
 * Score an ad with PreTestAds — plain x402 agent, no LLM, no account.
 *
 * Usage:
 *   npm install
 *   EVM_PRIVATE_KEY=0x... node score-ad.mjs https://example.com/ad.mp4 [windowSeconds]
 *
 * Costs $5.00 USDC on Base per run. Result in ~5-8 minutes.
 */
import { scoreAd } from "./lib/pretestads.mjs";

const [media, window = "15"] = process.argv.slice(2);
if (!media) {
  console.error("Usage: EVM_PRIVATE_KEY=0x... node score-ad.mjs <media_url_or_file> [3|5|15|30|60]");
  process.exit(1);
}

const result = await scoreAd(media, Number(window));
console.log(JSON.stringify(result, null, 2));
if (result.status === "complete") {
  console.log(`\nScore: ${result.score}/100 — ${result.label}`);
}

File: python/score_ad.py
#!/usr/bin/env python3
"""Score an ad with PreTestAds — plain x402 agent, no LLM, no account.

Usage:
    SOLANA_PRIVATE_KEY=<base58> python score_ad.py https://example.com/ad.mp4
    EVM_PRIVATE_KEY=0x...       python score_ad.py ~/Desktop/ad.mp4 --window 30

Costs $5.00 USDC per run (Solana or Base). Result in ~5-8 minutes.
"""
import argparse
import json

from pretestads import score_ad


def main():
    p = argparse.ArgumentParser(description=__doc__)
    p.add_argument("media", help="Public https:// URL or local file path (mp4/mov/avi/jpg/png/webp)")
    p.add_argument("--window", type=int, default=15, choices=[3, 5, 15, 30, 60],
                   help="Scoring window in seconds (default 15)")
    args = p.parse_args()

    result = score_ad(args.media, window_seconds=args.window)
    print(json.dumps(result, indent=2))
    if result.get("status") == "complete":
        print(f"\nScore: {result.get('score')}/100 — {result.get('label')}")


if __name__ == "__main__":
    main()
